### PR TITLE
Custom Armor Implemented! (VIPs keep their global rewards)

### PR DESCRIPTION
**Pull request by olliestanley**
SpoutPlugin: https://github.com/SpoutDev/SpoutPlugin/pull/70
SpoutcraftAPI: https://github.com/SpoutDev/SpoutcraftAPI/pull/36

### CODE REVIEW

**Comment by olliestanley** [comment]
Yes this is mainly @Wolftein's code, I have permission from him to use it in this specific context, and there are a few changes.

@ZNickq the packet is used in GenericCustomArmor#getPacketType (override from GenericCustomItem) so it is necessary.

File: src/minecraft/net/minecraft/src/RenderPlayer.java [hunk 1]
```diff
@@ -46,21 +48,38 @@ protected int setArmorModel(EntityPlayer par1EntityPlayer, int par2, float par3)
...
+                        return -1;
+                    }
+                } else {
+                    return -1;
+                }
+			} else {
+			    return -1;
 			}
+			//Spout End
+            ModelBiped var7 = par2 == 2?this.modelArmor:this.modelArmorChestplate;
```
Comment (ZNickq) [inline src/minecraft/net/minecraft/src/RenderPlayer.java]: Formatting...

Comment (olliestanley) [inline src/minecraft/net/minecraft/src/RenderPlayer.java]: Lol I use too many different formatting profiles... VP is spaces, CraftFire is spaces, then I forgot Spout is tabs :/. Will fix tomorrow.

File: src/minecraft/org/spoutcraft/client/packet/PacketType.java [hunk 2]
```diff
@@ -84,7 +84,8 @@
 	PacketSlotClick(59, PacketSlotClick.class),
 	PacketWaypoint(60, PacketWaypoint.class),
 	PacketParticle(61, PacketParticle.class),
-	PacketAccessory(62, org.spoutcraft.client.player.accessories.PacketAccessory.class);
+	PacketAccessory(62, org.spoutcraft.client.player.accessories.PacketAccessory.class),
+	PacketGenericArmor(63, PacketGenericArmor.class);
```
Comment (ZNickq) [inline src/minecraft/org/spoutcraft/client/packet/PacketType.java:88]: This isn't right, you don't need that packet at all, GenericCustomArmor should be a packet by itself, like GenericCustomItem :P

Comment (olliestanley) [inline src/minecraft/org/spoutcraft/client/packet/PacketType.java:88]: https://github.com/SpoutDev/Spoutcraft/blob/master/src/main/java/org/spoutcraft/client/packet/PacketGenericItem.java

Also see the rest of the PacketGeneric\* in that package.

Comment (ZNickq) [inline src/minecraft/org/spoutcraft/client/packet/PacketType.java:88]: BLeah, the names ar edifferent, seems like PacketGenericItem in SPN is called PacketCustomItem in SPC... will take a few mins to figure out what's going on :/

**Comment by ZNickq** [comment]
Alright, these look fine now :)
Once the first RB is out just rebase, and if everything is good in testing i'll pull them immediately ^^

**Comment by olliestanley** [comment]
I can't rebase SpoutPlugin because as I said I cannot amend commits. Also I have not grasped the concept of rebasing without making all the commits say

'Authored by ZNickq'
'Committed by DziNeIT' or something like that. :P

**Comment by olliestanley** [comment]
I'll let @Wolftein do these seeing as he is back on PR terms with you guys :P

**Pull request closed without merging**